### PR TITLE
homework submission week3

### PR DESCRIPTION
**Pull request by hnnhack**
* testing from postman must activate Header “Content-type: application/json”

http://localhost:3000/register post
{
	  "name": "niyas",
	  "username": "niyas",
	  "password": "niyas"
}
http://localhost:3000/login post
{
  "username": "niyas",
  "password": "niyas"
}
http://localhost:3000/logout get

http://localhost:3000/createList post
{
  "description": "yearly",
  "reminder_date": "2020-10-15 15:00:05"
}
http://localhost:3000/deleteList post
{
  "lid": 6,
  "description": "yearly"
}
http://localhost:3000/createItem post
{
  "lid": 7,
  "description": "play footbal"
}
http://localhost:3000/deleteItem post
{
  "lid": 7,
  "description": "play footbal"
}
http://localhost:3000/itemCompleted post
{
  "id": 10,
  "lid": 7
}

### CODE REVIEW

**Comment by roxana-lafuente** [comment]
> 
> 
>     * testing from postman must activate Header “Content-type: application/json”
> 
> 
> http://localhost:3000/register post
> {
> "name": "niyas",
> "username": "niyas",
> "password": "niyas"
> }
> http://localhost:3000/login post
> {
> "username": "niyas",
> "password": "niyas"
> }
> http://localhost:3000/logout get
> 
> http://localhost:3000/createList post
> {
> "description": "yearly",
> "reminder_date": "2020-10-15 15:00:05"
> }
> http://localhost:3000/deleteList post
> {
> "lid": 6,
> "description": "yearly"
> }
> http://localhost:3000/createItem post
> {
> "lid": 7,
> "description": "play footbal"
> }
> http://localhost:3000/deleteItem post
> {
> "lid": 7,
> "description": "play footbal"
> }
> http://localhost:3000/itemCompleted post
> {
> "id": 10,
> "lid": 7
> }

Great documentation, usually we add this in a README.md or similar file so that it doesn't get lost in the comments of github :)

File: Week3/homework/router.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        return response.redirect('/');
+      }
+      request.session.message = 'username already registered';
+      return response.redirect('/');
+    } catch (error) {
+      console.log(error.message);
+    }
+  })
+
+  .post('/signin', validator.login, async (request, response) => {
```
Comment (roxana-lafuente) [inline Week3/homework/router.js]: Great error handling!

File: Week3/homework/router.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        request.session.message = `Welcome ${users.name}`;
+        return response.redirect('/');
+      }
+      response.json('Please sign in / sign up');
+    } catch (error) {
+      console.log(error.message);
+    }
+  })
+
+  .get('/signout', async (request, response) => {
```
Comment (roxana-lafuente) [inline Week3/homework/router.js]: I would personally also use POST for logout too but amazing idea to provide these login and logout functionality :)

**Comment by hnnhack** [comment]
Oh! 
I had mistake with sql syntax in query file that **....where items.uid** must be **....where lists.uid**.
By the way I will add README file.
Thank you very much for your all kinds of efforts for review and etc.

**Pull request closed without merging**